### DISCUSSION TITLE
MultilineTextInput: Scroll to end of text

### DISCUSSION THREAD

**Discussion by MarkusPiotrowski** [beeware/toga#1722] [Q&A] [answered]
Does one have some control about the view of a MultilineTextInput field? Can I programmatically scroll down to the end of the text?

Edit: `scroll_to_bottom` and `scroll_to_top` have been implemented in version 0.3.0.

**Comment by freakboy3742** [comment, marked as answer]
Not at present. 

The Table and Tree widgets have a `scroll_to_top/bottom/row()` API; I suspect a similar API would work for MultilineTextInput as well. If someone were to submit a PR implementing it, I doubt there would be much argument on accepting it.

**Reply by MarkusPiotrowski** [reply]
I found a way to implement this (`scroll_to_bottom` and probably also `scroll_to_top`) for Windows (`toga_winforms\widgets\multilinetextinput.py`). Is it necessary for a PR to implement it for all supported platforms? What can I do, how should I proceed? 
I also have a MacBook and a Linux laptop available, so possibly I could supply implementations for these platforms later, too.

**Reply by freakboy3742** [reply]
You don't have to have a fully functioning implementation for every platform; however, if you're adding a new method that backend APIs will be required to implement, you do need to define a stub of that API on every backend. 

If you add any new backend methods to the `dummy` backend first, all the actual platform backends will start failing their tests. You can then populate the Winforms backend with a real implementation, and put in a "not implemented" implementation on other platforms (see [`Table.clear()` on winforms](https://github.com/beeware/toga/blob/main/winforms/src/toga_winforms/widgets/table.py#L121) for an example of how this is done).
